h2o-py/h2o/explanation/_explain.py

```diff
@@ -1327,21 +1332,24 @@ def ice_plot(
                 plt.plot(tmp[encoded_col], tmp["mean_response"], color=colors[i],
                          label="{}th Percentile".format(i * 10))
 
-        tmp = NumpyFrame(
-            model.partial_plot(
-                frame,
-                cols=[column],
-                plot=False,
-                targets=target,
-                nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
-            )[0]
-        )
-        if is_factor:
-            plt.scatter(factor_map(tmp.get(encoded_col)), tmp["mean_response"], color="k",
-                        label="Partial Dependence")
-        else:
-            plt.plot(tmp[encoded_col], tmp["mean_response"], color="k", linestyle="dashed",
-                     label="Partial Dependence")
+        if show_pdp:
+            tmp = NumpyFrame(
+                model.partial_plot(
+                    frame,
+                    cols=[column],
+                    plot=False,
+                    targets=target,
+                    nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
+                )[0]
+            )
+            if show_pdp:
+                encoded_col = tmp.columns[0]
```

Comment:
This check looks redundant, probably not needed

Reply:
[e536529](https://github.com/h2oai/h2o-3/pull/6016/commits/e53652901d9472e4597b350ca3a27c6668465e88)